pkg/execution/state/state.go

```diff
@@ -163,13 +164,52 @@ type Metadata struct {
 	// DisableImmediateExecution is used to tell the SDK whether it should
 	// disallow immediate execution of steps as they are found.
 	DisableImmediateExecution bool `json:"disableImmediateExecution,omitempty"`
+
+	// SpanID is the spanID used for this function run.
+	SpanID string `json:"sid"`
+	// QueuedAt shows the time the function was queued
+	QueuedAt int64 `json:"queuedAt"`
+	// StartedAt shows the time the function started execution
+	StartedAt int64 `json:"startedAt,omitempty"`
+
+	// The delay duration for the function to start caused by the system
+	// nano second precision
+	SystemDelay int64 `json:"sysdelay"`
```

Comment:
Why are these stored in a state's metadata?  Shouldn't this be per-step and per-queue item?

Reply:
This was needed prior because I was trying to annotate the function's delay before it starts.
It has changed to annotating the first step as the start of the function, so it's no longer necessary. I'll delete it.